code/lib/builder-vite/src/list-stories.ts

```diff
@@ -1,25 +1,27 @@
 import * as path from 'path';
 import slash from 'slash';
-import { glob } from 'glob';
+import * as glob from 'glob';
 import { normalizeStories } from '@storybook/core-common';
 
 import type { Options } from '@storybook/types';
 import { normalizePath } from 'vite';
 
 export async function listStories(options: Options) {
-  return (
-    await Promise.all(
-      normalizeStories(await options.presets.apply('stories', [], options), {
-        configDir: options.configDir,
-        workingDir: options.configDir,
-      }).map(({ directory, files }) => {
-        const pattern = path.join(directory, files);
-        const absolutePattern = path.isAbsolute(pattern)
-          ? pattern
-          : path.join(options.configDir, pattern);
+  const raw = await options.presets.apply('stories', [], options);
+  const normalizeOptions = {
+    configDir: options.configDir,
+    workingDir: options.configDir,
+  };
+  const normalized = await Promise.all(
+    normalizeStories(raw, normalizeOptions).map(async ({ directory, files }) => {
+      const pattern = path.join(directory, files);
+      const absolutePattern = path.isAbsolute(pattern)
+        ? pattern
+        : path.join(options.configDir, pattern);
 
-        return glob(slash(absolutePattern), { follow: true });
-      })
-    )
-  ).reduce((carry, stories) => carry.concat(stories.map(normalizePath)), []);
+      return glob.sync(slash(absolutePattern), { follow: true });
```

Comment:
So I discovered (by console.logging) that the output of `glob` (in contrast to what the type definitions say) is actually returning a Class instance.

I can't understand why.
Using the sync method does work.

It'd be preferred if we could stay on an async API.

Reply:
I've tried a few alternate strategies as well:

```
const glob = require('glob');

const glob = await import('glob');

import * as glob from 'glob';
```

None of that made any difference, the only way I could find to get it to return an array was with the sync method.

Reply:
I could also try to prebundle the `glob` package, I haven't tried that yet.

Reply:
@ndelangen is `glob.sync` sync? (I would guess so). If so, should we drop the `Promise.all`?

It's not ideal not to do this in parallel though.

Reply:
Yes it is sync, indeed.

Reply:
What if we actually used the Glob class instead (and import it as intended)? Based on the docs, it looks like `new Glob(..).walk()` is the same as `glob(...)`, and is async.

https://github.com/isaacs/node-glob#gwalk